Cabal-syntax/src/Distribution/Utils/Path.hs

```diff
@@ -531,3 +536,20 @@ data Response
 --
 -- See Note [Symbolic paths] in Distribution.Utils.Path.
 data PkgConf
+
+-------------------------------------------------------------------------------
+
+-- * Windows utils
+
+-------------------------------------------------------------------------------
+
+-- | Sometimes we need to represent a Windows path (that might have been
+-- normalized) as a POSIX path, for example in URIs.
+asPosixPath :: FilePath -> FilePath
+asPosixPath =
+  map
+    ( \x -> case x of
+        _
+          | x == Windows.pathSeparator -> Posix.pathSeparator
+          | otherwise -> x
+    )
```

Comment:
Can the case with wildcard and guards be simplified? Any of [`-XLambdaCase`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/lambda_case.html#extension-LambdaCase), `map (\x -> if ... then ... else ...)` or a list comprehension would be a simplification, wouldn't it?

Reply:
Done ✔️